ios/FluentUI/Notification/NotificationView.swift

```diff
@@ -548,14 +459,63 @@ open class NotificationView: UIView {
 
     @objc private func handleActionButtonTap() {
         hide(animated: true)
-        action?()
+        state.actionButtonAction?()
     }
 
     @objc private func handleMessageTap() {
-        guard let messageAction = messageAction else {
+        guard let messageAction = state.messageButtonAction else {
             return
         }
         hide(animated: true)
         messageAction()
     }
 }
+
+public class MSFNotificationStateImpl: NSObject, ObservableObject, Identifiable, MSFNotificationState {
```

Comment:
I am sure these state implies are needed if you are going to rewrite this control in swift ui eventually. but is it really necessary just to tokenized the current color? feels like changing how you would update the notficationview initialization sequence is orthogonal issue